### DISCUSSION TITLE
the .desktop file generated by distrobox does not work

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [sodaliterocks/sodalite#73] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/98482469/220439343-d5dfa7e9-828b-417f-85ed-50cc37626f06.png)

I try to open it but nothing happens it seems the error has to do with terminal not being gnome terminal or konsole
I don't know if the error has to do with the desktop patheon or distrobox

**Comment by electricduck** [comment, marked as answer]
_Moved this to a discussion as it's not a bug related to the OS itself, but isn't an invalid issue either._

---

_(Disclaimer: this answer is based on my rudimentary knowledge of GNOME/Glib, some quick research, and my own personal experience)_

So, `Terminal=` is a bit of an odd one in a `.desktop` file as it behaves differently depending on the DE. DEs using glib &mdash; so GNOME and Pantheon &mdash; have a hardcoded list of terminals to look through to launch from, and it seems Pantheon Terminal is missing from that list. You can try this yourself by doing `gtk-launch fedora-37.desktop`: you'll see it spit out an error that it cannot find a terminal.

There's also the property in dconf at `org.gnome.desktop.applications.terminal exec` which is set to `io.elementary.terminal`, but it doesn't seem this does anything.

You have two workarounds, however:

#### 1. Launching Pantheon Terminal directly:
 * Adjust the `Exec` property to be `/usr/bin/io.elementary.terminal -x /usr/bin/distrobox enter fedora-37`
 * Adjust the `TryExec` property to be `/usr/bin/io.elementary.terminal`

#### 2. Bash wrapper

Create a file at `/usr/local/bin/gnome-terminal` and add the following contents:

```
#!/usr/bin/env bash
shift
io.elementary.terminal -x $@
```

Don't forget to set it as executable too (`sudo chmod +x /usr/local/bin/gnome-terminal`).

Since `/usr/local/bin/` is in your `$PATH`, `glib` will search for `gnome-terminal` in there too and should fire it up; this command acts as a stub to redirect it to Pantheon Terminal. I would hardcode this into Sodalite itself, but it will make it impossible to install GNOME Terminal afterwards, which some users might legitimately need.

---

I also noticed you have a few errors in your desktop file:
  * `Category` should be `Categories`
  * You've gone a dangling quote (`"`) after your categories

### Further Reading
 * https://unix.stackexchange.com/questions/333661/how-to-specify-which-terminal-is-used-by-gtk-launch
 * https://askubuntu.com/questions/76712/setting-nautilus-open-terminal-to-launch-terminator-rather-than-gnome-terminal/294430#294430

**Reply by Heus-Sueh** [reply]
the second option worked thanks :+1:

**Reply by Heus-Sueh** [reply]
for now I think it would be good to deliver this Bash wrapper along with the distro, because maybe someone will have a similar problem

**Reply by electricduck** [reply]
> for now I think it would be good to deliver this Bash wrapper along with the distro, because maybe someone will have a similar problem

I would, but as explained, it will cause problems if you need to install GNOME Terminal afterwards, since there would be a package collision and it would fail to install.

`Terminal=` in desktop files is a very rarely used anymore &mdash; I'm gonna be honest, I didn't end know it still worked! &mdash; so it's a fairly non-issue currently.
